### PR TITLE
JNG-5134 Support for test database cleanup

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5134" title="JNG-5134" target="_blank">TaskJNG-5134</a>  Support for test database cleanup
  </td></table>
  <br />

### CODE REVIEW

**Review by bencegelei — APPROVED** [review]

File: judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java [hunk 1]
```diff
@@ -95,24 +102,43 @@ public void teardownDatasource() {
...
+                    log.debug("Truncating table: %s (%s)".formatted(rdbmsTable.getName(), rdbmsTable.getSqlName()));
+                    statement.execute("TRUNCATE TABLE " + rdbmsTable.getSqlName() + " RESTART IDENTITY CASCADE;");
+                }
             } else if (dialect.equals(DIALECT_HSQLDB)) {
-                statement.execute("TRUNCATE SCHEMA PUBLIC RESTART IDENTITY AND COMMIT NO CHECK");
-                //statement.execute("DROP SCHEMA PUBLIC CASCADE");
+                for (RdbmsTable rdbmsTable : rdbmsUtils.getRdbmsTables().orElse(new BasicEList<>())) {
+                    log.info("Truncating table: %s (%s)".formatted(rdbmsTable.getName(), rdbmsTable.getSqlName()));
+                    statement.execute("TRUNCATE TABLE " + rdbmsTable.getSqlName() + " RESTART IDENTITY AND COMMIT NO CHECK");
+                }
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java]: All two statement works with same dataSet, i n that case the statement template can be a template for String.format and on that case does not copy the iterator.

File: judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java [hunk 2]
```diff
@@ -95,24 +102,43 @@ public void teardownDatasource() {
...
+        try (Connection connection = dataSource.getConnection(); Statement statement = connection.createStatement()) {
+            if (dialect.equals(DIALECT_POSTGRESQL) || dialect.equals(DIALECT_HSQLDB)) {
+                for (RdbmsTable rdbmsTable : rdbmsUtils.getRdbmsTables().orElse(new BasicEList<>())) {
+                    log.debug("Drop table: %s (%s)".formatted(rdbmsTable.getName(), rdbmsTable.getSqlName()));
+                    statement.execute("DROP TABLE " + rdbmsTable.getSqlName() + " CASCADE;");
+                }
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Could not drop tables", throwables);
+        }
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java]: Whu the dialect selection here? There is no other alternative.

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java]: [01b3021](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/242/commits/01b30214a4bec63182a30c9527cdd8ca8b092286)

File: judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java]: [01b3021](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/242/commits/01b30214a4bec63182a30c9527cdd8ca8b092286)

**Review by noherczeg — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Review by Madmat8 — APPROVED** [review]

File: judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java [hunk 3]
```diff
@@ -95,24 +103,38 @@ public void teardownDatasource() {
         }
     }
 
-    public void dropSchema() {
+    public void truncateTables(RdbmsModel rdbmsModel) {
+        RdbmsUtils rdbmsUtils = new RdbmsUtils(rdbmsModel.getResourceSet());
+        HashMap<String, String> statmentTemplate = new HashMap<>();
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java]: This hasmap I think have to be static in the begining of class. You can us ImmutableMap for it.

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java (hunk 3 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/common/src/main/java/hu/blackbelt/judo/runtime/core/jsl/fixture/JudoDatasourceFixture.java]: [bb3a87b](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/242/commits/bb3a87b315d5eff48012bff0e9765287ac0b52e8)

**Review by bencegelei — APPROVED** [review]

**Review by robertcsakany — APPROVED** [review]

**Pull request merged**